pkg/util/docker/dockerfile/builder/client.go

```diff
@@ -301,7 +303,8 @@ func (e *ClientExecutor) LoadImage(from string) (*docker.Image, error) {
 	for _, config := range auth {
 		// TODO: handle IDs?
 		// TODO: use RawJSONStream:true and handle the output nicely
-		if err = e.Client.PullImage(docker.PullImageOptions{Repository: from, OutputStream: e.Out, Tag: tag}, config); err == nil {
+		auth := docker.AuthConfiguration{Username: config.Username, ServerAddress: config.ServerAddress, Password: config.Password}
```

Comment:
don't shadow the `auth` var. should we be using the struct values assuming they are non-empty, or checking/calling `.Provider.LazyProvide()` on it? this is a weird struct...